roles/reproducer/tasks/push_code.yml

```diff
@@ -74,15 +74,35 @@
 - name: Push random code into the proper location
   vars:
     repo_base_dir: '/home/zuul/src'
-  when:
-    - not cifmw_reproducer_skip_fetch_repositories | default(false)
+    _cifmw_reproducer_all_repositories: "{{ cifmw_reproducer_repositories | default([]) }}"
   block:
+    - name: Expand cifmw_reproducer_repositories to pull code from ansible controller to controller-0
+      when:
+        - cifmw_reproducer_skip_fetch_repositories | default(false)
+        - zuul is defined
+        - zuul.items is defined
+        - zuul.projects is defined
+        # check that the project is not alredy in the user defined
+        # cifmw_reproducer_repositories value
+        - repo.value.name not in _user_sources
+      vars:
+        _user_sources: "{{ cifmw_reproducer_repositories | default([]) | map(attribute='src') }}"
+        _repo_entry:
+          src: "{{ ansible_user_dir }}/{{ repo.value.src_dir | regex_replace('/$', '') }}/"
+          dest: "/home/zuul/{{ repo.value.src_dir }}"
+      ansible.builtin.set_fact:
+        _cifmw_reproducer_all_repositories: "{{ _cifmw_reproducer_all_repositories  + [_repo_entry] }}"
```

Comment:
Is this properly overriding the variable declared here? Cause set_fact cannot change the value of a variable.